### ISSUE TITLE
Warning: unknown channel option SO_KEEPALIVE

### ISSUE THREAD

**Issue by lorien** [yidongnan/grpc-spring-boot-starter#624]
## The context

I want to use grpc server inside spring boot server. I want grpc server listens on unix domain socket.

## The question

I have upgraded to recent release of grpc-server-spring-boot-starter and added this config line to properties file:
grpc.server.address=unix:/path/to/backend-grpc.sock
It works OK but I keep seeing these lines in log file:

```22:09:03.656 [epollEventLoopGroup-2-1] WARN i.g.n.s.i.n.b.ServerBootstrap - Unknown channel option 'SO_KEEPALIVE' for channel '[id: 0xdf583dc3]'```

Could you please suggest how can I fix that issue? I do not want these lines appear in log file.

## The application's environment

Spring (boot): 2.5.7
grpc-java: 1.35.1
grpc-server-spring-boot-starter: 2.13.0.RELEASE
java: 17.0.1, 64bit

**Comment by ST-DDT** [comment]
The keepalive is set here:

https://github.com/yidongnan/grpc-spring-boot-starter/blob/aee1d1486973f06d1b7be3a030b3a2c309b671a4/grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/serverfactory/NettyGrpcServerFactory.java#L98-L107

Can you please check which of these properties causes this?
If not, it might take some time for me to fix this.

**Note:** You should probably update to a newer version of grpc/protobuf as a security advisory has been published for protobuf a few days ago. (The spring-boot update might be important as well, if you are using log4j2)
See also: https://github.com/yidongnan/grpc-spring-boot-starter/releases/tag/v2.13.1.RELEASE

**Comment by midnight-wonderer** [comment]
I was tracking down this issue today.  
It is such a common problem, I don't know why so few people are here.

### The Solution
I am years apart from you, but here is the solution anyway.
The warning is triggered by: [gRPC Netty (NettyServer.java#L222)](https://github.com/grpc/grpc-java/blob/7507a9ec06fc9490c5bb967c3afea953e9d1c98d/netty/src/main/java/io/grpc/netty/NettyServer.java#L222).

There is a way to override the option!!!
The override logic is at [NettyServer.java#L236](https://github.com/grpc/grpc-java/blob/7507a9ec06fc9490c5bb967c3afea953e9d1c98d/netty/src/main/java/io/grpc/netty/NettyServer.java#L236).

Basically, applying `.withChildOption(ChannelOption.SO_KEEPALIVE, null)` to `NettyServerBuilder` would do the trick.

Most people here just don't affected; they don't use Epoll.
